lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/ProcessCriCallbackHandler.java

```diff
@@ -155,61 +156,86 @@ public APIGatewayProxyResponseEvent handleRequest(
         } catch (ParseCriCallbackRequestException e) {
-
             return buildErrorResponse(
                     e,
                     HttpStatusCode.BAD_REQUEST,
-                    ErrorResponse.FAILED_TO_PARSE_CRI_CALLBACK_REQUEST);
+                    ErrorResponse.FAILED_TO_PARSE_CRI_CALLBACK_REQUEST,
+                    Level.ERROR);
         } catch (InvalidCriCallbackRequestException e) {
-            if (e.getErrorResponse() == ErrorResponse.NO_IPV_FOR_CRI_OAUTH_SESSION) {
-                LOGGER.error(LogHelper.buildErrorMessage(e.getErrorResponse()));
-                var pageOutput =
-                        StepFunctionHelpers.generatePageOutputMap(
-                                "error",
-                                HttpStatusCode.UNAUTHORIZED,
-                                PYI_TIMEOUT_RECOVERABLE_PAGE_ID);
-                var criOAuthSessionItem =
-                        criOAuthSessionService.getCriOauthSessionItem(callbackRequest.getState());
-                if (criOAuthSessionItem != null) {
-                    pageOutput.put(
-                            "clientOAuthSessionId", criOAuthSessionItem.getClientOAuthSessionId());
+            switch (e.getErrorResponse()) {
+                case NO_IPV_FOR_CRI_OAUTH_SESSION -> {
+                    LOGGER.warn(LogHelper.buildErrorMessage(e.getErrorResponse()));
+                    var pageOutput =
+                            StepFunctionHelpers.generatePageOutputMap(
+                                    "error",
+                                    HttpStatusCode.UNAUTHORIZED,
+                                    PYI_TIMEOUT_RECOVERABLE_PAGE_ID);
+                    var criOAuthSessionItem =
+                            criOAuthSessionService.getCriOauthSessionItem(
+                                    callbackRequest.getState());
+                    if (criOAuthSessionItem != null) {
+                        pageOutput.put(
+                                "clientOAuthSessionId",
+                                criOAuthSessionItem.getClientOAuthSessionId());
+                    }
+                    return ApiGatewayResponseGenerator.proxyJsonResponse(
+                            HttpStatusCode.UNAUTHORIZED, pageOutput);
+                }
+                case INVALID_OAUTH_STATE -> {
+                    LOGGER.warn(LogHelper.buildErrorMessage(e.getErrorResponse()));
+                    return ApiGatewayResponseGenerator.proxyJsonResponse(
+                            HttpStatusCode.BAD_REQUEST,
+                            StepFunctionHelpers.generatePageOutputMap(
+                                    "error",
+                                    HttpStatusCode.BAD_REQUEST,
+                                    PYI_ATTEMPT_RECOVERY_PAGE_ID));
+                }
+                case MISSING_IPV_SESSION_ID -> {
+                    LOGGER.warn(LogHelper.buildErrorMessage(e.getErrorResponse()));
```

Comment:
Should we log here, or it already gets logged by the caller? (and if so what about the default case too?)